hi-ml/src/health_ml/run_ml.py

```diff
@@ -0,0 +1,233 @@
+        model configuration object or the lightning container object is defined.
+        For models defined as lightning containers, this is the namespace of the container class defining the model.
+
+        :return: the namespace of the model configuraton object
+        """
+        return self.container.__class__.__module__
+
+    def set_run_tags_from_parent(self) -> None:
+        """
+        Set metadata for the run
+        """
+        assert PARENT_RUN_CONTEXT, "This function should only be called in a Hyperdrive run."
+        run_tags_parent = PARENT_RUN_CONTEXT.get_tags()
+        tags_to_copy = [
+            "tag",
+            "model_name",
+            "execution_mode",
+            "recovered_from",
+            "friendly_name",
+            "build_number",
+            "build_user",
+            RUN_RECOVERY_FROM_ID_KEY_NAME
+        ]
+        new_tags = {tag: run_tags_parent.get(tag, "") for tag in tags_to_copy}
+        new_tags[RUN_RECOVERY_ID_KEY] = create_run_recovery_id(run=RUN_CONTEXT)
+        new_tags[CROSSVAL_SPLIT_KEY] = str(self.container.crossval_split_index)
+        new_tags[EFFECTIVE_RANDOM_SEED_KEY_NAME] = str(self.container.get_effective_random_seed())
+        RUN_CONTEXT.set_tags(new_tags)
+
+    def set_multiprocessing_start_method(self) -> None:
+        """
+        Set the (PyTorch) multiprocessing start method.
+        """
+        if hasattr(self.container, "multiprocessing_start_method"):
+            method = self.container.multiprocessing_start_method
+        else:
+            method = MultiprocessingStartMethod.spawn
+        if is_windows():
+            if method != MultiprocessingStartMethod.spawn:
+                logging.warning(f"Cannot set multiprocessing start method to '{method.name}' "
+                                "because only 'spawn' is available in Windows")
+        else:
+            logging.info(f"Setting multiprocessing start method to '{method.name}'")
+            torch.multiprocessing.set_start_method(method.name, force=True)
+
+    def run(self) -> None:
+        """
+        Driver function to run a ML experiment. If an offline cross validation run is requested, then
+        this function is recursively called for each cross validation split.
+        """
+        self.setup()
+        # Get the AzureML context in which the script is running
+        if not self.is_offline_run and PARENT_RUN_CONTEXT is not None:
+            logging.info("Setting tags from parent run.")
+            self.set_run_tags_from_parent()
+
+        # Set data loader start method
+        self.set_multiprocessing_start_method()
+
+        # do training
+        with logging_section("Model training"):
+            _, storing_logger = model_train(container=self.container,
+                                            num_nodes=self.experiment_config.num_nodes)
+            self.storing_logger = storing_logger
+
+        RUN_CONTEXT.log(name="Train epochs", value=self.container.max_epochs)
```

Comment:
why are we logging this, actually? we are logging hyperparams, and number of epochs is there.